sympy/tensor/tests/test_indexed.py

```diff
@@ -353,3 +353,12 @@ def test_indexed_is_constant():
     assert not A[1+2*i, k].is_constant(i)
     assert A[1+2*i, k].is_constant(j)
     assert not A[1+2*i, k].is_constant(k)
+
+
+def test_issue_12533():
+    d = IndexedBase('d')
+    assert IndexedBase(range(5)) == Array([0, 1, 2, 3, 4])
+    assert d[0].subs(Symbol("d"), range(5)) == 0
```

Comment:
> And honestly, the fact that IndexedBase stores a Symbol as its arg instead of just being atomic itself is a bit surprising to me. Any reason for that?

It is already like that in master.

> or really the line https://github.com/sympy/sympy/pull/12556/files#diff-f91a3a19a7672c48a33a63796c8e9b09R380 doesn't make sense to me.

Maybe it can just return `label` or the sympified `_sympify(label)`.